packages/rest/src/body-parsers/body-parser.json.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {RequestBodyParserOptions} from '../types';
+
+import {json} from 'body-parser';
+import {inject} from '@loopback/context';
+import {Request} from '../..';
```

Comment:
I am worrying about circular dependencies. Can re-arrange type definitions so that deep files like this one can import `Request` from a non-index file, e.g. `../types`?

Reply:
> The code in individual built-in parser classes seems to be duplicated a lot, have you considered using a Strategy or Template design pattern to extract a shared base class?

The size of each class is fairly small as most of the common logic have been pushed into the helper class. Introducing a new base class seems to overkill and add complexity, IMO.

Reply:
@raymondfeng your answer above seems to be related to a different comment, could you PTAL?